File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/NodeExpressMongooseTypescriptServerCodegen.java
package org.openapitools.codegen.languages;

import org.openapitools.codegen.*;
import io.swagger.models.properties.ArrayProperty;
import io.swagger.models.properties.MapProperty;
import io.swagger.models.properties.Property;
import io.swagger.models.parameters.Parameter;

import java.io.File;
import java.util.*;

import org.apache.commons.lang3.StringUtils;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class NodeExpressMongooseTypescriptServerCodegen extends DefaultCodegen implements CodegenConfig {
    public static final String PROJECT_NAME = "projectName";

    private final Logger LOGGER = LoggerFactory.getLogger(NodeExpressMongooseTypescriptServerCodegen.class);

    public CodegenType getTag() {
        return CodegenType.SERVER;
    }

    public String getName() {
        return "node-express-mongoose-typescript";
    }

    public String getHelp() {
        return "Generates a node-express-mongoose-typescript server.";
    }

    public NodeExpressMongooseTypescriptServerCodegen() {
        super();

        outputFolder = "generated-code" + File.separator + "node-express-mongoose-typescript";
        modelTemplateFiles.put("model.mustache", ".zz");
        apiTemplateFiles.put("api.mustache", ".zz");
        embeddedTemplateDir = templateDir = "node-express-mongoose-typescript";
        apiPackage = "Apis";
        modelPackage = "Models";
        supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
        // TODO: Fill this out.
    }
}

File: modules/openapi-generator/src/main/kotlin/org/openapitools/codegen/languages/NodeExpressMongooseTypescriptServerCodegen.kt
package org.openapitools.codegen.languages

import org.openapitools.codegen.*
import io.swagger.models.properties.ArrayProperty
import io.swagger.models.properties.MapProperty
import io.swagger.models.properties.Property
import io.swagger.models.parameters.Parameter
import java.io.File
import java.util.*
import org.apache.commons.lang3.StringUtils
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class NodeExpressMongooseTypescriptServerCodegen : DefaultCodegen(), CodegenConfig {
  companion object {
    const val PROJECT_NAME = "projectName"
  }

  private val LOGGER: Logger = LoggerFactory.getLogger(NodeExpressMongooseTypescriptServerCodegen::class.java)

  override fun getTag(): CodegenType {
    return CodegenType.SERVER
  }

  override fun getName(): String {
    return "node-express-mongoose-typescript"
  }

  override fun getHelp(): String {
    return "Generates a node-express-mongoose-typescript server."
  }

  init {
    outputFolder = "generated-code${File.separator}node-express-mongoose-typescript"
    modelTemplateFiles["model.mustache"] = ".ts"
    apiTemplateFiles["api.mustache"] = ".ts"
    embeddedTemplateDir = "node-express-mongoose-typescript"
    templateDir = "node-express-mongoose-typescript"
    apiPackage = "Apis"
    modelPackage = "Models"
    supportingFiles.add(SupportingFile("README.mustache", "", "README.md"))
    // TODO: Fill this out.
  }
}

